main.py

```
@@ -0,0 +1,105 @@
+
+
+async def process_video(path: str, yaml_file: str):
+    from ruamel.yaml import YAML
+    r_yaml = YAML()
+    r_yaml.preserve_quotes = True
+    with open(yaml_file, 'r') as file:
+        data = r_yaml.load(file)
+    data['video_path'] = path
+    with open(yaml_file, 'w') as file:
+        r_yaml.dump(data, file)
+    result = subprocess.run([sys.executable, "dancepose/scripts/run_pose.py"], capture_output=True, text=True)
```

Ruff (0.14.3)

`subprocess` call: check for execution of untrusted input

(S603)